spec/features/admin/applications/show_spec.rb

```diff
@@ -0,0 +1,121 @@
+
+    @application_pet_1 = ApplicationPet.create!(application_id: @application_1.id, pet_id: @pet_1.id)
+    @application_pet_2 = ApplicationPet.create!(application_id: @application_1.id, pet_id: @pet_2.id)
+    @application_pet_3 = ApplicationPet.create!(application_id: @application_2.id, pet_id: @pet_1.id)
+  end
+
+  it 'has a button to approve or reject a pet on an application' do
+    visit "/admin/applications/#{@application_1.id}"
+
+    within "##{@pet_1.id}" do
+      expect(page).to have_button("Approve Application")
+      expect(page).to have_button("Reject Application")
+    end
+    within "##{@pet_2.id}" do
+      expect(page).to have_button("Approve Application")
+      expect(page).to have_button("Reject Application")
+    end
+  end
+
+  it 'can accept the application for a pet and see an indicator that that application was accepted' do
+    visit "/admin/applications/#{@application_1.id}"
+
+    within "##{@pet_1.id}" do
+      click_on "Approve Application"
+    end
+
+    expect(current_path).to eq("/admin/applications/#{@application_1.id}")
+
+    within "##{@pet_1.id}" do
+      expect(page).not_to have_button("Approve Application")
+      expect(page).not_to have_button("Reject Application")
+      expect(page).to have_content("Application Approved")
+    end
+  end
+
+  it 'can reject the application for a pet and see an indicator that that application was rejected' do
+    visit "/admin/applications/#{@application_1.id}"
+
+    within "##{@pet_1.id}" do
+      click_on "Reject Application"
+    end
+
+    expect(current_path).to eq("/admin/applications/#{@application_1.id}")
+
+    within "##{@pet_1.id}" do
+      expect(page).not_to have_button("Approve Application")
+      expect(page).not_to have_button("Reject Application")
+      expect(page).to have_content("Application Rejected")
+    end
+  end
+
+  it 'will not effect a separate application for the same pet if the pet is approved on a separate application' do
+    visit "/admin/applications/#{@application_1.id}"
+
+    within "##{@pet_1.id}" do
+      click_on "Approve Application"
+    end
+
+    visit "/admin/applications/#{@application_2.id}"
+
+    within "##{@pet_1.id}" do
+      expect(page).to have_button("Approve Application")
+      expect(page).to have_button("Reject Application")
+      expect(page).not_to have_content("Application Accepted")
+    end
+  end
+
+  it 'will not effect a separate application for the same pet if the pet is rejected on a separate application' do
+    visit "/admin/applications/#{@application_1.id}"
+
+    within "##{@pet_1.id}" do
+      click_on "Reject Application"
+    end
+
+    visit "/admin/applications/#{@application_2.id}"
+
+    within "##{@pet_1.id}" do
+      expect(page).to have_button("Approve Application")
+      expect(page).to have_button("Reject Application")
+      expect(page).not_to have_content("Application Rejected")
+    end
+  end
+end
```

Comment:
This is an amazing test sweet, seriously. 🤩 

Everything is here... you have `within` blocks, `before :each`, `to have`, `not_to have`, logical separation of assertions. These tests are great. Don't forget that you can also tuck your `visit` in the `before :each` as well.